### PR TITLE
Create integration tests using a sample test App

### PR DESCRIPTION
**Pull request by jmrossy**
Increase test coverage by creating a sample application using this lib and test it using Eth Mainnet.

### CODE REVIEW

File: src/test/App.test.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    render(<App />)
+    // Check that block timestamp is correctly retrieved from block
+    const timestamp1 = await waitFor(() => screen.getByTestId('blockTimestamp'), { timeout: 15_000 /* 15 seconds */ })
+    expect(timestamp1 && timestamp1?.textContent).toBeTruthy()
+    const value1 = parseInt(timestamp1.textContent!, 10) * 1000
+    const now1 = Date.now()
+    expect(now1 - value1).toBeLessThan(MAX_BLOCK_AGE)
+
+    // Wait for an updated block timestamp
+    await sleep(10_000 /* 10 seconds */)
```
Comment (moodysalem) [inline src/test/App.test.tsx:34]: this is gonna be pretty flaky isn't it? block times are 12 seconds on mainnet

File: src/test/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  return blockNumber
+}
+
+export function useCurrentBlockTimestamp(blockNumber: number | undefined): string | undefined {
+  const contract = useContract()
+  const result = useSingleCallResult(1, blockNumber, contract, 'getCurrentBlockTimestamp')
+  return result.result?.[0]?.toString()
+}
+
+export function useMaxTokenBalance(blockNumber: number | undefined): string | undefined {
```
Comment (moodysalem) [inline src/test/hooks.ts:39]: weird name, wasn't sure what it did before i read impl

File: src/test/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  const ERC20Interface = new utils.Interface(ERC20_ABI)
+  const results = useMultipleContractSingleData(
+    1,
+    blockNumber,
+    [USDC_ADDRESS, USDT_ADDRESS],
```
Comment (moodysalem) [inline src/test/hooks.ts:44]: off topic: one thing i want to better understand is what happens when these inputs are not memoized, does it attach/unattach listeners every time the hook is called? and then what would the side effect be?

**Review by moodysalem — COMMENTED** [review]
kinda strange to have test code in `src/test` and also tests adjacent to the source code

File: src/test/App.test.tsx (hunk 1 above)
Comment (jmrossy) [reply src/test/App.test.tsx:34]: Worked well so far but I'll increase the times for reliability

**Comment by moodysalem** [comment]
i think the only thing i would change is maybe putting this in its own `integration-tests` folder with its own separate script

File: src/test/hooks.ts (hunk 2 above)
Comment (jmrossy) [reply src/test/hooks.ts:39]: It's just a random example of using useMultipleContractSingleData

Comment (jmrossy) [reply src/test/hooks.ts]: But open to suggestions

**Review by moodysalem — APPROVED** [review]

**Pull request merged**